packages/medusa/src/strategies/batch-jobs/price-list/import.ts

```diff
@@ -110,7 +123,13 @@ class PriceListImportStrategy extends AbstractBatchJobStrategy {
 
     // Validate that PriceList exists
     const priceListId = batchJob.context.price_list_id as string
-    await this.priceListService_.withTransaction(manager).retrieve(priceListId)
+    if (this.featureFlagRouter_.isFeatureEnabled(MedusaV2Flag.key)) {
```

Comment:
question: Same as riqwan pr, can we create a prop on the class to store the if the flag is enbabled and in the constructor we can assign the result of this?